src/script/error/BaseError.ts

```diff
@@ -17,38 +17,41 @@
  *
  */
 
+enum BASE_ERROR_TYPE {
+  INVALID_PARAMETER = 'INVALID_PARAMETER',
+  MISSING_PARAMETER = 'MISSING_PARAMETER',
+  UNKNOWN = 'UNKNOWN',
+}
+
 export class BaseError extends Error {
-  constructor(name, type, message) {
+  type: BASE_ERROR_TYPE | string;
+
+  constructor(name: string, type: BASE_ERROR_TYPE | string, message: string) {
     super();
 
-    this.name = this.constructor.name;
     this.stack = new Error().stack;
+    this.name = this.constructor.name;
+    this.message = message || BaseError.MESSAGE[type];
+    this.type = type ? type : BaseError.TYPE.UNKNOWN;
```

Comment:
```suggestion
    this.type = type || BaseError.TYPE.UNKNOWN;
```

Reply:
After talking to @bennyn we decided now that this is migrated to TypeScript let's just go with the required parameters so there would be no more needing to fall back to default things. I'll add a few commits and let you know to please take a look again.

Reply:
Alright!